pkg/sql/unsafesql/unsafesql_test.go

```diff
@@ -305,3 +305,77 @@ func TestPanickingSQLFormat(t *testing.T) {
 		string(result),
 	)
 }
+
+func TestClusterSettingOverride(t *testing.T) {
```

Comment:
Is it possible to add this test case to an existing unit test? Just wondering because the compound costs of `StartServer` in every single unit test begin to add up over time. Not a huge deal, but nice to avoid.

Reply:
We could, we would just need to add one layer of nesting to the suite, which should be trivial.

It could be a datadriven test, I personally prefer the table test structure as is done here.